src/pretix/control/views/checkin.py

```diff
@@ -206,14 +226,20 @@ def post(self, request, *args, **kwargs):
                         'web': True
                     }, user=request.user)
                     checkin_created.send(op.order.event, checkin=ci)
+            return 'ok', request.POST.get('returnquery')
 
-            messages.success(request, _('The selected tickets have been marked as checked in.'))
+    def get_success_message(self, value):
+        if value[0] == 'reverted':
+            return _('The selected check-ins have been reverted.')
+        else:
+            return _('The selected tickets have been marked as checked in.')
```

Comment:
This will also return the success message for checked-in for tickets marked as checked out. In line 229 we need a third option next to `'ok'`, or instead of ok, use something like `checked-in` and `checked-out`.

Reply:
Sorry, I did not wait and just added the third option. I was too lazy to turn this into a dict of messages, but I think with three options that is kind of still okay. If there will be a fourth option, we should consider turning the if-elif-else block into a dict-lookup.